internal/operator-controller/rukpak/convert/generate.go

```diff
@@ -0,0 +1,203 @@
+package convert
+
+import (
+	"cmp"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	registrybundle "github.com/operator-framework/operator-registry/pkg/lib/bundle"
+
+	"github.com/operator-framework/operator-controller/internal/operator-controller/rukpak/util"
+)
+
+type UniqueNameGenerator func(string, interface{}) (string, error)
+
+type Options struct {
+	InstallNamespace    string
+	TargetNamespaces    []string
+	UniqueNameGenerator UniqueNameGenerator
+}
+
+type ResourceGenerator func(rv1 *RegistryV1, opts Options) ([]client.Object, error)
+
+func (g ResourceGenerator) GenerateResources(rv1 *RegistryV1, opts Options) ([]client.Object, error) {
+	return g(rv1, opts)
+}
+
+func ChainedResourceGenerator(resourceGenerators ...ResourceGenerator) ResourceGenerator {
+	return func(rv1 *RegistryV1, opts Options) ([]client.Object, error) {
+		//nolint:prealloc
+		var renderedObjects []client.Object
+		for _, generator := range resourceGenerators {
+			objs, err := generator(rv1, opts)
+			if err != nil {
+				return nil, err
+			}
+			renderedObjects = append(renderedObjects, objs...)
+		}
+		return renderedObjects, nil
+	}
+}
+
+func BundleDeploymentGenerator(rv1 *RegistryV1, opts Options) ([]client.Object, error) {
+	if rv1 == nil {
+		return nil, fmt.Errorf("bundle cannot be nil")
+	}
+	objs := make([]client.Object, 0, len(rv1.CSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs))
+	for _, depSpec := range rv1.CSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		annotations := util.MergeMaps(rv1.CSV.Annotations, depSpec.Spec.Template.Annotations)
+		annotations["olm.targetNamespaces"] = strings.Join(opts.TargetNamespaces, ",")
+		depSpec.Spec.Template.Annotations = annotations
+
+		// Hardcode the deployment with RevisionHistoryLimit=1 (something OLMv0 does, not sure why)
```

Comment:
Looks like we have the answer for this [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/dfd0b2bea85038d3c0d65348bc812d297f16b8d2/pkg/controller/install/deployment.go#L177-L180) right?

Reply:
Fixed =D